From a09b9abae63fa4ab25244533a8bfafc6cc5039d6 Mon Sep 17 00:00:00 2001
From: Rahul T R <r-ravikumar@ti.com>
Date: Wed, 9 Oct 2024 23:35:20 +0530
Subject: [PATCH] arch: arm: dts: k3-j784s4-*/k3-am69-*: Update memory map for
 PSDK RTOS

PSDK RTOS uses different memory map to accomodate larger FWs and
remote core memory needs

Signed-off-by: Rahul T R <r-ravikumar@ti.com>
---
 arch/arm/dts/k3-am69-sk.dts                 |   2 +
 arch/arm/dts/k3-j784s4-evm.dts              |   2 +
 arch/arm/dts/k3-j784s4-rtos-memory-map.dtsi | 335 ++++++++++++++++++++
 3 files changed, 339 insertions(+)
 create mode 100644 arch/arm/dts/k3-j784s4-rtos-memory-map.dtsi

diff --git a/arch/arm/dts/k3-am69-sk.dts b/arch/arm/dts/k3-am69-sk.dts
index c352530e3eb..c56d69efe84 100644
--- a/arch/arm/dts/k3-am69-sk.dts
+++ b/arch/arm/dts/k3-am69-sk.dts
@@ -1334,3 +1334,5 @@
 	phys = <&serdes0_usb_link>;
 	phy-names = "cdns3,usb3-phy";
 };
+
+#include "k3-j784s4-rtos-memory-map.dtsi"
diff --git a/arch/arm/dts/k3-j784s4-evm.dts b/arch/arm/dts/k3-j784s4-evm.dts
index afd84a6dfae..89e0d4df98f 100644
--- a/arch/arm/dts/k3-j784s4-evm.dts
+++ b/arch/arm/dts/k3-j784s4-evm.dts
@@ -1696,3 +1696,5 @@
 	tx-num-evt = <0>;
 	rx-num-evt = <0>;
 };
+
+#include "k3-j784s4-rtos-memory-map.dtsi"
diff --git a/arch/arm/dts/k3-j784s4-rtos-memory-map.dtsi b/arch/arm/dts/k3-j784s4-rtos-memory-map.dtsi
new file mode 100644
index 00000000000..4f8ce9d9f59
--- /dev/null
+++ b/arch/arm/dts/k3-j784s4-rtos-memory-map.dtsi
@@ -0,0 +1,335 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Vision-apps: device-tree overlay
+ *
+ * Copyright (C) 2019 - 2022 Texas Instruments Incorporated - http://www.ti.com/
+ */
+
+#include <dt-bindings/soc/ti,sci_pm_domain.h>
+
+&mcu_r5fss0_core0_memory_region {
+	status = "disabled";
+};
+
+&mcu_r5fss0_core0_dma_memory_region {
+	status = "disabled";
+};
+
+&mcu_r5fss0_core1_dma_memory_region {
+	status = "disabled";
+};
+
+&mcu_r5fss0_core1_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss0_core0_dma_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss0_core0_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss0_core1_dma_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss0_core1_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss1_core0_dma_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss1_core0_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss1_core1_dma_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss1_core1_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss2_core0_dma_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss2_core0_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss2_core1_dma_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss2_core1_memory_region {
+	status = "disabled";
+};
+
+&c71_0_dma_memory_region {
+	status = "disabled";
+};
+
+&c71_0_memory_region {
+	status = "disabled";
+};
+
+&c71_1_dma_memory_region {
+	status = "disabled";
+};
+
+&c71_1_memory_region {
+	status = "disabled";
+};
+
+&c71_2_dma_memory_region {
+	status = "disabled";
+};
+
+&c71_2_memory_region {
+	status = "disabled";
+};
+
+&c71_3_dma_memory_region {
+	status = "disabled";
+};
+
+&c71_3_memory_region {
+	status = "disabled";
+};
+
+&reserved_memory {
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	vision_apps_mcu_r5fss0_core0_dma_memory_region: vision-apps-r5f-dma-memory@a0000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa0000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_mcu_r5fss0_core0_memory_region: vision-apps-r5f-memory@a0100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa0100000 0x00 0x00f00000>;
+		no-map;
+	};
+	vision_apps_mcu_r5fss0_core1_dma_memory_region: vision-apps-r5f-dma-memory@a1000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa1000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_mcu_r5fss0_core1_memory_region: vision-apps-r5f-memory@a1100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa1100000 0x00 0x00f00000>;
+		no-map;
+	};
+	vision_apps_main_r5fss0_core0_dma_memory_region: vision-apps-r5f-dma-memory@a2000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa2000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_main_r5fss0_core0_memory_region: vision-apps-r5f-memory@a2100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa2100000 0x00 0x01f00000>;
+		no-map;
+	};
+	vision_apps_main_r5fss0_core1_dma_memory_region: vision-apps-r5f-dma-memory@a4000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa4000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_main_r5fss0_core1_memory_region: vision-apps-r5f-memory@a4100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa4100000 0x00 0x00f00000>;
+		no-map;
+	};
+	vision_apps_main_r5fss1_core0_dma_memory_region: vision-apps-r5f-dma-memory@a5000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa5000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_main_r5fss1_core0_memory_region: vision-apps-r5f-memory@a5100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa5100000 0x00 0x00f00000>;
+		no-map;
+	};
+	vision_apps_main_r5fss1_core1_dma_memory_region: vision-apps-r5f-dma-memory@a6000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa6000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_main_r5fss1_core1_memory_region: vision-apps-r5f-memory@a6100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa6100000 0x00 0x00f00000>;
+		no-map;
+	};
+	vision_apps_main_r5fss2_core0_dma_memory_region: vision-apps-r5f-dma-memory@a7000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa7000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_main_r5fss2_core0_memory_region: vision-apps-r5f-memory@a7100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa7100000 0x00 0x00f00000>;
+		no-map;
+	};
+	vision_apps_main_r5fss2_core1_dma_memory_region: vision-apps-r5f-dma-memory@a8000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa8000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_main_r5fss2_core1_memory_region: vision-apps-r5f-memory@a8100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa8100000 0x00 0x00f00000>;
+		no-map;
+	};
+	vision_apps_rtos_ipc_memory_region: vision-apps-rtos-ipc-memory-region@ac000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xac000000 0x00 0x03000000>;
+		no-map;
+	};
+	vision_apps_memory_region: vision-apps-dma-memory@af000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xaf000000 0x00 0x03000000>;
+		no-map;
+	};
+	vision_apps_c71_0_dma_memory_region: vision-apps-c71-dma-memory@b2000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb2000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_c71_0_memory_region: vision-apps-c71_0-memory@b2100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb2100000 0x00 0x00504000>;
+		no-map;
+	};
+	vision_apps_c71_1_dma_memory_region: vision-apps-c71_1-dma-memory@b4000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb4000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_c71_1_memory_region: vision-apps-c71_1-memory@b4100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb4100000 0x00 0x00504000>;
+		no-map;
+	};
+	vision_apps_c71_2_dma_memory_region: vision-apps-c71_2-dma-memory@b6000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb6000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_c71_2_memory_region: vision-apps-c71_2-memory@b6100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb6100000 0x00 0x00504000>;
+		no-map;
+	};
+	vision_apps_c71_3_dma_memory_region: vision-apps-c71_3-dma-memory@b8000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb8000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_c71_3_memory_region: vision-apps-c71_3-memory@b8100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb8100000 0x00 0x00504000>;
+		no-map;
+	};
+	vision_apps_core_heaps_lo: vision-apps-core-heap-memory-lo@ba000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xba000000 0x00 0x04000000>;
+		no-map;
+	};
+	vision_apps_main_r5fss0_core0_shared_memory_queue_region: vision-apps-r5f-virtual-eth-queues@be000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xbe000000 0x00 0x00800000>;
+		no-map;
+	};
+	vision_apps_main_r5fss0_core0_shared_memory_bufpool_region: vision-apps-r5f-virtual-eth-buffers@be800000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xbe800000 0x00 0x01800000>;
+		no-map;
+	};
+	vision_apps_core_heaps_hi: vision-apps-core-heap-memory-hi@880000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x08 0x80000000 0x00 0x80000000>;
+		no-map;
+	};
+	vision_apps_shared_region: vision_apps_shared-memories {
+		compatible = "dma-heap-carveout";
+		reg = <0x09 0x00000000 0x00 0x3c000000>;
+	};
+	linux,cma {
+		status = "disabled";
+	};
+	linux_cma_region: linux-cma-buffers@980000000 {
+		compatible = "shared-dma-pool";
+		reusable;
+		reg = <0x09 0x80000000 0x00 0x70000000>;
+		linux,cma-default;
+	};
+};
+
+&mcu_r5fss0_core0 {
+	memory-region = <&vision_apps_mcu_r5fss0_core0_dma_memory_region>,
+			<&vision_apps_mcu_r5fss0_core0_memory_region>;
+};
+
+&mcu_r5fss0_core1 {
+	memory-region = <&vision_apps_mcu_r5fss0_core1_dma_memory_region>,
+			<&vision_apps_mcu_r5fss0_core1_memory_region>;
+};
+
+&main_r5fss0_core0 {
+	memory-region = <&vision_apps_main_r5fss0_core0_dma_memory_region>,
+			<&vision_apps_main_r5fss0_core0_memory_region>,
+			<&vision_apps_main_r5fss0_core0_shared_memory_queue_region>,
+			<&vision_apps_main_r5fss0_core0_shared_memory_bufpool_region>;
+};
+
+&main_r5fss0_core1 {
+	memory-region = <&vision_apps_main_r5fss0_core1_dma_memory_region>,
+			<&vision_apps_main_r5fss0_core1_memory_region>;
+};
+
+&main_r5fss1_core0 {
+	memory-region = <&vision_apps_main_r5fss1_core0_dma_memory_region>,
+			<&vision_apps_main_r5fss1_core0_memory_region>;
+};
+
+&main_r5fss1_core1 {
+	memory-region = <&vision_apps_main_r5fss1_core1_dma_memory_region>,
+			<&vision_apps_main_r5fss1_core1_memory_region>;
+};
+
+&main_r5fss2_core0 {
+	memory-region = <&vision_apps_main_r5fss2_core0_dma_memory_region>,
+			<&vision_apps_main_r5fss2_core0_memory_region>;
+};
+
+&main_r5fss2_core1 {
+	memory-region = <&vision_apps_main_r5fss2_core1_dma_memory_region>,
+			<&vision_apps_main_r5fss2_core1_memory_region>;
+};
+
+&c71_0 {
+	memory-region = <&vision_apps_c71_0_dma_memory_region>,
+			<&vision_apps_c71_0_memory_region>;
+};
+
+&c71_1 {
+	memory-region = <&vision_apps_c71_1_dma_memory_region>,
+			<&vision_apps_c71_1_memory_region>;
+};
+
+&c71_2 {
+	memory-region = <&vision_apps_c71_2_dma_memory_region>,
+			<&vision_apps_c71_2_memory_region>;
+};
+
+&c71_3 {
+	memory-region = <&vision_apps_c71_3_dma_memory_region>,
+			<&vision_apps_c71_3_memory_region>;
+};
-- 
2.34.1

